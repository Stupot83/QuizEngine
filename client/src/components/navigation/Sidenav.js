import React from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { logoutUser } from "../../actions/actionsForAuthentication";
import clsx from "clsx";
import { makeStyles } from "@material-ui/core/styles";
import Drawer from "@material-ui/core/Drawer";
import CssBaseline from "@material-ui/core/CssBaseline";
import Toolbar from "@material-ui/core/Toolbar";
import List from "@material-ui/core/List";
import Divider from "@material-ui/core/Divider";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import HomeIcon from "@material-ui/icons/Home";
import ExitToAppIcon from "@material-ui/icons/ExitToApp";
import Logo from "../../../src/Logo.png";
import "../../sass/Sidenav.scss";

const drawerWidth = 240;

const useStyles = makeStyles(theme => ({
    root: {
        display: "flex"
    },
    Toolbar: {
        marginTop: "-64px"
    },
    menuButton: {
        marginRight: theme.spacing(2)
    },
    hide: {
        display: "none"
    },
    drawer: {
        width: drawerWidth,
        flexShrink: 0
    },
    drawerPaper: {
        width: drawerWidth,
        border: "1px solid #070707"
    },
    drawerHeader: {
        display: "flex",
        alignItems: "center",
        padding: theme.spacing(0, 1),
        ...theme.mixins.toolbar,
        justifyContent: "flex-end"
    }
}));

const Sidenav = props => {
    const classes = useStyles();
    const [open, setOpen] = React.useState(false);

    const handleDrawerOpen = () => {
        setOpen(true);
    };

    const handleDrawerClose = () => {
        setOpen(false);
    };

    const onLogoutClick = e => {
        e.preventDefault();
        props.logoutUser(props.history);
        window.location.href = "/";
    };

    return (
        <div className={classes.root}>
            <CssBaseline />
            <Toolbar className={classes.Toolbar}>
                <IconButton
                    color="inherit"
                    aria-label="open drawer"
                    onClick={handleDrawerOpen}
                    edge="start"
                    className={clsx(classes.menuButton, open && classes.hide)}
                >
                    <MenuIcon id="hamburger" className="Hamburger_icon" fontSize="large" />
                </IconButton>
            </Toolbar>
            <Drawer
                id="Drawer"
                className={classes.drawer}
                variant="persistent"
                anchor="left"
                open={open}
                classes={{
                    paper: classes.drawerPaper
                }}
            >
                <div className={classes.drawerHeader}>
                    <img id="logo" className="Logo" src={Logo} alt="Logo" />
                    <IconButton onClick={handleDrawerClose}>
                        <ChevronLeftIcon id="chevron" fontSize="large" />
                    </IconButton>
                </div>
                <Divider />
                <List>
                    <ListItem button component={Link} to="/display">
                        <ListItemIcon>
                            <HomeIcon id="homeIcon" fontSize="large" />
                        </ListItemIcon>
                        <ListItemText id="homeHeader" primary="Home" />
                    </ListItem>
                    <ListItem button onClick={onLogoutClick}>
                        <ListItemIcon>
                            <ExitToAppIcon id="signOutIcon" fontSize="large" />
                        </ListItemIcon>
                        <ListItemText id="signOutHeader" primary="Sign-out" />
                    </ListItem>
                    <Divider />
                </List>
            </Drawer>
        </div>
    );
};

const mapStateToProps = state => ({
    auth: state.auth
});

export default connect(mapStateToProps, { logoutUser })(withRouter(Sidenav));
