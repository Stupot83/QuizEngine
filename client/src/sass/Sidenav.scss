.Hamburger_icon {
 color: #070707;
}

.MuiDrawer-paper {
 background-color: #fff;
background-image:
linear-gradient(90deg, transparent 79px, #abced4 79px, #abced4 81px, transparent 81px),
linear-gradient(#eee .1em, transparent .1em);
background-size: 100% 1.2em;
}

.MuiSvgIcon-fontSizeLarge {
 color: #070707 !important;
}

.MuiTypography-root {
 padding-left: 30px;
}

.MuiListItemIcon-root {
 padding-left: 20px;
 color: #070707 !important;
}

.MuiTypography-body1 {
 color: #070707 !important;
 font-weight: bold !important;
}

.Logo {
 text-align: center;
 width: 80%;
}

@media (max-width: 600px) {
 .Hamburger_icon {
  margin-top: 8px;
 }
}