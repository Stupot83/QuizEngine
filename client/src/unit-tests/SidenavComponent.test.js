import React from "react";
import { shallow } from "enzyme";
import Sidenav from "../components/navigation/Sidenav";
import QuizEngineStore from "../quizEngineStore";
import { Provider } from "react-redux";

const mockStore = QuizEngineStore;

describe("Sidenav component", () => {
    it("should render correctly in debug mode", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <Sidenav debug />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <Sidenav />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});
