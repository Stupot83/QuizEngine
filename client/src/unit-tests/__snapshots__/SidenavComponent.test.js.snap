// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidenav component should render correctly 1`] = `ShallowWrapper {}`;

exports[`Sidenav component should render correctly in debug mode 1`] = `ShallowWrapper {}`;
